coconut/configuration/configuration.go

```diff
@@ -332,133 +315,94 @@ func History(cfg *configuration.ConsulSource, cmd *cobra.Command, args []string,
-	if err != nil {
-		return "", errors.New(fmt.Sprintf("Unable to identify timestamp"))
+	if len(args) != 3 {
+		return errors.New(fmt.Sprintf("Accepts exactly 3 args but received %d", len(args))), invalidArgs
 	}
-	tm := time.Unix(timeStampAsInt, 0)
-	return  tm.Format(timeFormat), nil
-}
 
-// Method to return the latest timestamp for a specified component & entry
-// If no keys were passed an error and code exit 3 will be returned
-func GetLatestTimestamp(keys []string, component string, entry string)(timestamp string, err error, code int) {
-	keyPrefix := componentsPath + component + "/" + entry
-	if len(keys) == 0 {
-		err = errors.New(fmt.Sprintf("No keys found"))
-		return "", err, emptyData
+	if !isInputSingleValidWord(args[0]) || !isInputSingleValidWord(args[1]) && args[2] != "" {
+		return errors.New(fmt.Sprintf(invalidArgsErrMsg)), invalidArgs
 	}
 
-	var maxTimeStamp uint64
-	for _, key := range keys {
-		componentTimestamp, err := strconv.ParseUint(strings.TrimPrefix(key, keyPrefix + "/"), 10, 64)
-		if err == nil {
-			if componentTimestamp > maxTimeStamp  {
-				maxTimeStamp = componentTimestamp
-			}
-		}
+	component, entry, filePath := args[0], args[1], args[2]
+
+	fileParts := strings.Split(filePath, ".")
+	extension := ""
+	if len(fileParts) > 1 {
+		extension = fileParts[len(fileParts)-1]
 	}
-	return strconv.FormatUint(maxTimeStamp, 10), nil, nonZero
-}
 
-// Method to return a list of components, entries or entries with latest timestamp
-// If no keys were passed an error and code exit 3 will be returned
-func GetListOfComponentsAndOrWithTimestamps(keys []string, keyPrefix string, useTimestamp bool)([]string, error, int) {
-	if len(keys) == 0 {
-		return []string{},  errors.New(fmt.Sprintf("No keys found")), emptyData
+	if !isFileExtensionValid(extension) &&  useExtension == "" {
+		return errors.New(fmt.Sprintf("Extension of the file should be: JSON, YAML, INI or TOML  or for a different extension " +
+			"please use flag '-f/--format' and specify the extension.", )), invalidArgs
+	} else if useExtension != ""  {
+		extension = strings.ToUpper(useExtension)
 	}
 
-	var components []string
-	componentsSet := make(map[string]string)
+	keys, err := cfg.GetKeysByPrefix("", "")
+	if err != nil {
+		return  err, connectionError
+	}
 
-	for _, key := range keys {
-		componentsFullName := strings.TrimPrefix(key, keyPrefix)
-		componentParts := strings.Split(componentsFullName, "/")
-		componentTimestamp := componentParts[len(componentParts) - 1]
-		if useTimestamp {
-			componentsFullName = strings.TrimSuffix(componentsFullName, "/" +componentTimestamp)
-		} else {
-			componentsFullName = componentParts[0]
+	components := getComponentsMapFromKeysList(keys)
+	componentExist := components[component]
+	if !componentExist &&  !useNewComponent {
+		componentMsg := ""
+		for key, _ := range components {
+			componentMsg += "\n-" + key
 		}
+		return errors.New(fmt.Sprintf("Component `" + component + "` does not exist. " +
```

Comment:
Let's not use backticks here, either color or nothing. The backticks can potentially make sense in Cobra command doc strings because those are exported as Markdown, but not in error messages.